k8sclient/rest_client.go

```diff
@@ -0,0 +1,31 @@
+package k8sclient
+
+import (
+	"github.com/giantswarm/microerror"
+	"k8s.io/client-go/rest"
+)
+
+func NewRestClient(config Config, group, version string) (rest.Interface, error) {
+	err := config.Validate()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if config.Group == "" {
+		return nil, microerror.Maskf(invalidConfigError, "Group must not be empty")
+	}
```

Comment:
Why is this not part of the validate method?

Reply:
From `Config` struct:

```go
	// Group is required only for NewRestClient.
	Group string
	// Version is required only for NewRestClient.
	Version string
```

Other clients (non-rest) specify those values on their own. E.g. `ApiextensionsV1beta1` sets group to `"apiextensions.k8s.io"` and version to `"v1beta1"`. When you create your own REST client you have to provide those values, otherwise `rest.RESTClientFor` cries.

With that you can do things like: 	`client.Get().Namespace("...").Resource("kvms").DoRaw()`.
Next steps is to work on decoding/encoding, so we can do `client.Get().Namespace("...").Resource("kvms").Watch()`.